
class Exception(msg)
{
  var message = msg;

  fn __toString()
  {
    return message;
  }

  return this;
}

fn outter()
{
  fn middle(mp1, mp2)
  {
    print("middle:", mp1, mp2);
  }

  try
  {
    print("outter");
    throw Exception("E:herp derp");
  }
  catch e
  {
    print("outter:catch");
    throw e;
  }
}



var aa = "none";

if aa
{
  print("> cond is true");
}
else
{
  print("> cond is false");
}


try
{
  print("do stuff");
  outter();
}
catch e
{
  print("Got exception", e);
  //throw e;
}
